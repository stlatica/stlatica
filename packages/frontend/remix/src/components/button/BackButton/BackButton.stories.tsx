import { BackButton } from "./BackButton";

import type { Story, StoryDefault } from "@ladle/react";

export default {
  title: "button/BackButton",
} satisfies StoryDefault;

export const Story1: Story = () => {
  return <BackButton />;
};
