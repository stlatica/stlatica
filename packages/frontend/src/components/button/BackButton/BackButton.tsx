import { UnstyledButton } from "@mantine/core";
import React from "react";
import { MdArrowBack } from "react-icons/md";

type BackButtonProps = {
  // children: React.ReactNode;
  readonly onClick?: () => void;
};

/**
 *
 */
export const BackButton: React.FC<BackButtonProps> = ({ onClick }) => {
  return (
    <UnstyledButton onClick={onClick}>
      <MdArrowBack size="3em" />
    </UnstyledButton>
  );
};
