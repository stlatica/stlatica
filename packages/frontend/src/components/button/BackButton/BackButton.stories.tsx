import React from "react";

import { BackButton } from "./BackButton";

import type { Story, StoryDefault } from "@ladle/react";

export default {
  title: "components/button/BackButton",
} satisfies StoryDefault;

export const Story1: Story = () => {
  return <BackButton />;
};
