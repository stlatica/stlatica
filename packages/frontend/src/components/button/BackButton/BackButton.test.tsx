import { render } from "@testing-library/react";
import React from "react";
import { describe, expect, test } from "vitest";

import { Story1 } from "./BackButton.stories";

describe("snapshot", () => {
  test("snapshot", () => {
    const { asFragment } = render(<Story1 />);
    expect(asFragment()).toMatchSnapshot();
  });
});
