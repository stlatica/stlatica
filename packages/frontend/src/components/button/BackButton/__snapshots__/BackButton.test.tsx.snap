// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot > snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center"
  >
    <button
      class="border-2 border-black"
      type="button"
    >
      <svg
        fill="currentColor"
        height="3em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="3em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
